snuba/admin/migrations_policies.py

```diff
@@ -0,0 +1,68 @@
+from __future__ import annotations
+
+from functools import wraps
+from typing import Any, Callable, Dict
+
+from flask import Response, jsonify, make_response, request
+from werkzeug.routing import BaseConverter
+
+from snuba import settings
+from snuba.migrations.policies import (
+    AllMigrationsPolicy,
+    MigrationPolicy,
+    NoMigrationsPolicy,
+    NonBlockingMigrationsPolicy,
+)
+from snuba.migrations.runner import MigrationKey
+
+
+class MigrationActionConverter(BaseConverter):
+    regex = r"(?:run|reverse)"
+
+
+def check_migration_perms(f: Callable[..., Response]) -> Callable[..., Response]:
+    """
+    A wrapper decorator applied to endpoints handling migrations. It checks that
+    the migration group is in the ADMIN_ALLOWED_MIGRATION_GROUPS setting and checks
+    that the migration has the appropriate run/reverse policy
+    """
+
+    @wraps(f)
+    def check_group_perms(*args: Any, **kwargs: Any) -> Response:
+        policy_map = {
+            "AllMigrationsPolicy": AllMigrationsPolicy(),
+            "NoMigrationsPolicy": NoMigrationsPolicy(),
+            "NonBlockingMigrationsPolicy": NonBlockingMigrationsPolicy(),
+        }
```

Comment:
We now have a concept of registered classes to resolve a class by name. 
https://github.com/getsentry/snuba/blob/master/snuba/utils/registered_class.py#L46

You may want to remove this boilerplate. 
These are only three and quite static so no big deal, up to you.

Reply:
fixed